crates/interpreter/src/syntax.rs

```diff
@@ -546,6 +557,45 @@ impl CvtOp {
     }
 }
 
+#[cfg(feature = "threads")]
+macro_rules! impl_atomic_op {
+    ($u:ident) => {
+        paste::paste! {
+            impl AtomicUnOp {
+                pub fn $u(&self, ptr: *mut $u, c: $u) -> $u {
+                    let x = unsafe { [< Atomic $u:upper >]::from_ptr(ptr) };
+                    match self {
+                        AtomicUnOp::Add => x.fetch_add(c, Ordering::SeqCst),
+                        AtomicUnOp::Sub => x.fetch_sub(c, Ordering::SeqCst),
+                        AtomicUnOp::And => x.fetch_and(c, Ordering::SeqCst),
+                        AtomicUnOp::Or => x.fetch_or(c, Ordering::SeqCst),
+                        AtomicUnOp::Xor => x.fetch_xor(c, Ordering::SeqCst),
+                        AtomicUnOp::Xchg => x.swap(c, Ordering::SeqCst),
+                    }
+                }
+            }
+            impl AtomicBinOp {
+                pub fn $u(&self, ptr: *mut $u, c: $u, c2: $u) -> $u {
+                    let x = unsafe { [< Atomic $u:upper >]::from_ptr(ptr) };
+                    match x.compare_exchange(c2, c, Ordering::SeqCst, Ordering::SeqCst) {
```

Comment:
Why are the arguments swapped? This feels counter-intuitive and I'm not sure it's correct.